### PR TITLE
TOC, tidb-in-kubernetes: re-org TiDB in K8s to move all topics to one place

### PR DESCRIPTION
**Pull request by dcalvin**
### What is changed, added or deleted? 

This PR re-organized the TOC so that all TiDB in Kubernetes topics are aggregated in one place.

**IMPORTANT**: Some PRs on docs are still to be merged, so it is normal if some links in the TOC is not working. DNM until I have confirmed that all files are properly merged.

I will raise another PR for 3.0 when this is merged.

### Which version does your change affect? 

  Dev

### Checklist 

- [x] Add a new file to `TOC.md`
- [ ] No Tab spaces anywhere 
- [ ] Leave a blank line both before and after a code block
- [ ] Keep the first level heading consistent with `title` in metadata
- [ ] Use *four* spaces for each level of indentation except that in `TOC.md`

### CODE REVIEW

**Comment by dcalvin** [comment]
@TomShawn FYI. You can refer to the re-org TOC for links that are used in the TiDB Operator Overview document.

**Comment by TomShawn** [comment]
> @TomShawn FYI. You can refer to the re-org TOC for links that are used in the TiDB Operator Overview document.

Thanks!

File: dev/TOC.md [hunk 1]
```diff
@@ -347,12 +325,46 @@
...
+    - [DinD](tidb-in-kubernetes/get-started/deploy-tidb-from-kubernetes-dind.md)
+    - [GKE](tidb-in-kubernetes/get-started/deploy-tidb-from-kubernetes-gke.md)
+    - [Minikube](tidb-in-kubernetes/get-started/deploy-tidb-from-kubernetes-minikube.md)
+  + Deploy
+    - [Prerequisites for TiDB in Kubernetes](tidb-in-kubernetes/deploy/prerequisites.md)
+    - [TiDB in AWS EKS](tidb-in-kubernetes/deploy/aws-eks.md)
+    - [TiDB in GCP](how-to/tidb-in-kubernetes/gcp-gke.md)
+    - [TiDB in Alibaba Cloud](how-to/tidb-in-kubernetes/alibaba-cloud.md)
+    - [TiDB in General Kubernetes](tidb-in-kubernetes/deploy/general-kubernetes.md)
+    - [Deploy TiDB Operator](tidb-in-kubernetes/deploy/tidb-operator.md)
```
Comment (aylei) [inline dev/TOC.md]: I'd suggest changing the order:
- Prerequisites
- Deploy TiDB Operator
- TiDB in General Kubernetes
- Public clouds

File: dev/TOC.md [hunk 2]
```diff
@@ -350,12 +328,47 @@
...
+  + Maintain
+    - [Destroy a TiDB cluster](tidb-in-kubernetes/maintain/delete-cluster.md)
+    - [Maintain a Hosting Kubernetes Node](tidb-in-kubernetes/maintain/kubernetes-node.md)
+    - [Backup and Restore](tidb-in-kubernetes/maintain/backup-and-restore.md)
+    - [Collect Logs](tidb-in-kubernetes/maintain/log-collecting.md)
+    - [Automatic Failover](tidb-in-kubernetes/maintain/auto-failover.md)
+  - [Scale](tidb-in-kubernetes/scale-in-kubernetes.md)
+  + Upgrade
+    - [TiDB Cluster](tidb-in-kubernetes/upgrade/tidb-cluster.md)
+    - [TiDb Operator](tidb-in-kubernetes/upgrade/tidb-operator.md)
```
Comment (aylei) [inline dev/TOC.md]:
```suggestion
    - [TiDB Operator](tidb-in-kubernetes/upgrade/tidb-operator.md)
```

**Comment by aylei** [comment]
Things are well organized now, excellent works!

File: dev/TOC.md [hunk 3]
```diff
@@ -350,12 +328,47 @@
...
+    - [TiDB Cluster](tidb-in-kubernetes/upgrade/tidb-cluster.md)
+    - [TiDB Operator](tidb-in-kubernetes/upgrade/tidb-operator.md)
+  + Reference
+    + Configuration
+      - [TiDB Cluster](tidb-in-kubernetes/reference/configuration/tidb-cluster.md)
+      - [Backup](tidb-in-kubernetes/reference/configuration/backup.md)
+      - [Local PV](tidb-in-kubernetes/reference/configuration/local-pv.md)
+    + Tools
+      - [tkctl](tidb-in-kubernetes/reference/tools/tkctl.md)
+      - [Tools in Kubernetes](tidb-in-kubernetes/reference/tools/in-kubernetes.md)
```
Comment (yikeke) [inline dev/TOC.md]: Where is this `Tools in Kubernetes` document?

Comment (dcalvin) [reply dev/TOC.md]: Not merged yet. Tom is reviewing it.

Comment (yikeke) [reply dev/TOC.md]: Ok.

Comment (TomShawn) [reply dev/TOC.md]: Review finished.

**Pull request merged**